pkgs/development/compilers/cudatoolkit/stdenv.nix

```diff
@@ -0,0 +1,10 @@
+{ overrideCC
+, wrapCCWith
+, stdenv
+, compatibleStdenv
+}:
+
+overrideCC stdenv (wrapCCWith rec {
+  cc = compatibleStdenv.cc.cc;
+  libcxx = stdenv.cc.cc.lib;
```

Comment:
To clarify: would it be possible to build against gcc11's `libcxx` using `overrideCC stdenv compatibleStdenv.cc` and then link/patchelf with gcc12's `libcxx` in `buildInputs`? Here we would be taking advantage of backwards compatibility between `libcxx` versions.

Reply:
> can we simply set this to null

We need exactly the opposite: we want this to be gcc12's libstdc++. If we use libstdc++ from gcc11 we get the mistmatch as soon as we import any C++ library from the normal nixpkgs

Reply:
Could we not build with the standard gcc11 toolchain and then link/patchelf against gcc12's `libcxx`?

Reply:
Yes, theoretically, we could. In fact, that was pennae's original suggestion. I think this would be more error-prone: we can forget to remove a reference to gcc11 in one place, we can forget to add a reference to gcc12 in another place, and we'd also have to do that in some hooks. I absolutely despise hooks, though I usually have to submit to them.

The approach in this PR, however, means that gcc11's libstdc++ never enters our build environment, unless a user accidentally adds it. The latter is a possibility, and I myself have made this mistake several times already. I think we want to run sanity checks, not patchelf

Reply:
Ahh ok, I gotcha! Thanks for explaining!

Reply:
not a blocker for this PR, but would it be possible to include a simple check for gcc11's libstdc++ in `preBuild` or something like that?

Reply:
There's a number arguments that `pkgs/build-support/cc-wrapper/default.nix` accepts. I would guess that both extraBuildCommands and `{extra,native}{Tools,Packages} must allow setting up with a bit of bash hackery. I have not yet explored them

Reply:
As I mentioned, I'd also like `cudaStdenv` to include a working `nvcc`, so there's another reason to learn more about these inputs